app/services/services/tab-webcontents/main.ts

```diff
@@ -212,35 +213,49 @@ export class TabWebContentsServiceImpl extends TabWebContentsService implements
-      return fromEvent(wc, 'new-window', (event, url, _, disposition, options) => ({ event, url, disposition, options }))
-        .subscribe(async ({ event, url, disposition, options }) => {
-          if (disposition === 'new-window') {
-            if (options) {
-              options.fullscreen = false;
-            }
-          } else if (disposition === 'foreground-tab' && String(url).startsWith('about:blank')) {
-            // Gmail PDF hack. Will download a printed PDF from thumbnail
-            // EDIT: not only Gmail or just PDF but most of the URL link on a Google app with overriden User Agent
-            // also falls here
-            event.preventDefault();
-            const guest = await handleHackGoogleAppsURLs(event, options);
-
-            if (guest) { // not a download
-              const newWindowUrl = guest.webContents.getURL();
-              if (newWindowUrl.startsWith('about:blank')) {
-                // if popup is still blank after 2 seconds, we show it to let it finish
-                guest.show();
-              } else {
-                guest.close();
-                // otherwise dispatch the current URL of the guest window into our URLRouter
-                await provider.dispatchUrl(newWindowUrl, wc.id);
-              }
-            }
-          } else {
-            event.preventDefault();
-            await provider.dispatchUrl(url, wc.id);
+
+      wc.setWindowOpenHandler((details: HandlerDetails) => {
+
+        log.debug('WindowOpen', details, process.type);
+
+        if (details.disposition === 'new-window') {
+          provider.dispatchUrl(details.url, wc.id, DEFAULT_BROWSER_BACKGROUND);
+          return { action: 'deny' };
+        }
+        else if (details.disposition === 'background-tab') {
+          provider.dispatchUrl(details.url, wc.id, DEFAULT_BROWSER);
+          return { action: 'deny' };
+        }
+
+        //vk: 2023.09.29 don't know how to verify this hack
+        //    will fix it later
+
+        // else if (disposition === 'foreground-tab' && String(url).startsWith('about:blank')) {
+        //   // Gmail PDF hack. Will download a printed PDF from thumbnail
+        //   // EDIT: not only Gmail or just PDF but most of the URL link on a Google app with overriden User Agent
+        //   // also falls here
+        //   const guest = await handleHackGoogleAppsURLs(event, options);
+
+        //   if (guest) { // not a download
+        //     const newWindowUrl = guest.webContents.getURL();
+        //     if (newWindowUrl.startsWith('about:blank')) {
+        //       // if popup is still blank after 2 seconds, we show it to let it finish
+        //       guest.show();
+        //     } else {
+        //       guest.close();
+        //       // otherwise dispatch the current URL of the guest window into our URLRouter
+        //       await provider.dispatchUrl(newWindowUrl, wc.id);
+        //     }
+        //   }
+        // }
```

Comment:
I don't remembder the exact issue, but clicking on a PDF file on a thread of mails wasn't working as expected in Station (it did nothing I think)

Reply:
yes. Now I see the problem. Previous hack doesn't work now.
I'll create an issue, so we won't lose it.